lib/callbacks/onegini_custom_registration_callback.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:flutter/services.dart';
+import 'package:onegini/constants/constants.dart';
+
+import '../onegini.dart';
+
+/// A callback to respond to the init and finish of customRegistrationActions.
+class OneginiCustomRegistrationCallback {
+    Future<void> submitSuccessAction(String identityProviderId, String? token) async {
+        await Onegini.instance.channel.invokeMethod(
+            Constants.submitCustomRegistrationSuccessAction,
+            <String, dynamic>{
```

Comment:
Why not `<String, String?>` ?

Reply:
went for dynamic since in
`<String, dynamic>{
                'identityProviderId': identityProviderId,
                'token': token
            }`

providerId would be String while token would be String?; If we would go with String? for both cases it would maybe make it look like providerId is also of type String?

Reply:
I'm not sure I'm following 😅 I suggested going for `String?` just for `token`, not `identityProviderId`

Reply:
its for the map defenition, as far as i am aware when you define the structure you define all the types that are allowed within that map. So the options as far as i know are:
<String, String?>{ 'identityProviderId': identityProviderId, 'token': token }
or 
<String, dynamic>{ 'identityProviderId': identityProviderId, 'token': token }

But the first option says that identityProviderId is of type String? while it would be String

Reply:
Isn't using `String?` instead of `dynamic` restricts allowed data types? `dynamic` accepts all the types; floats, ints, including nullable string, where `String?` accepts only String and nullable strings. 
So `String?` covers all the values that can get into map - String and nullable String where as `dynamic` accepts both String types + different types.

Reply:
yes it restricts more the datatypes; but in the case of <String, String?> this would mean that both token and identityProviderId are of type String?, while identityProviderId is of type String which why i thought it was better to highlight that there are different datatypes in the map, by allowing "everything" using dynamic to force people to think. But i think i am fine with both at this point i think.